Sources/ClientRuntime/Networking/Http/Middlewares/ContentLengthMiddleware.swift

```diff
@@ -22,8 +29,12 @@ public struct ContentLengthMiddleware<OperationStackOutput: HttpResponseBinding>
         case .stream(let stream):
             if let length = stream.length {
                 input.headers.update(name: "Content-Length", value: String(length))
+            } else if !requiresLength && unsignedPayload {
+                // only for HTTP/1.1 requests, will be removed in all HTTP/2 requests
+                input.headers.update(name: "Transfer-Encoding", value: "Chunked")
             } else {
-                input.headers.update(name: "Transfer-Encoded", value: "Chunked")
+                let errorMessage = unsignedPayload ? "operation requires length" : "sigv4 requires length"
```

Comment:
I think the error messages should have more details to make debugging easier for customers, e.g., "Stream operation [OPERATION NAME] requires content length." Or "SigV4 signing requires content length for [OPERATION NAME]."

Reply:
I think in this case adding the operation name would be redundant since ContentLengthMiddleware() is code-generated into relevant operations. I took the error messages as specified in the SEP through I agree more verbose error messages are almost always better. Take a look at the codegen test below to see what I mean and let me know what we think. If we want the operation name it would have to be passed in as an additional argument to the middleware.

Reply:
I don't think it'll be redundant because even though the middleware is code-generated into each operation, if multiple operations were called on services in an application written by SDK end-user, they would have to track down inside which operation this middleware failed due to insufficient error message.

The name of the operation is actually passed within the middleware context object `context` to the `handle` operation, and you can access it easily using `context.getOperation()` - no code-gen stage input is required.

Reply:
Awesome I agree, thanks!